disnake/http.py

```diff
@@ -450,24 +1115,43 @@ async def close(self) -> None:
 
     # login management
 
-    async def static_login(self, token: str) -> user.User:
-        # Necessary to get aiohttp to stop complaining about session creation
-        self.__session = aiohttp.ClientSession(
-            connector=self.connector, ws_response_class=DiscordClientWebSocketResponse
-        )
-        old_token = self.token
-        self.token = token
+    async def static_login(self, auth: str) -> user.User:
+        # # Necessary to get aiohttp to stop complaining about session creation
+        # self.__session = aiohttp.ClientSession(
+        #     connector=self.connector, ws_response_class=DiscordClientWebSocketResponse
+        # )
+        # old_token = self.token
+        # self.token = token
+
+        # TODO: Change this? This is literally just fetching /users/@me AKA "Get Current User", and is totally
+        #  usable with OAuth2. This doesn't actually have anything to do with logging in.
+        self._set_default_auth(auth)
 
         try:
             data: user.User = await self.request(Route("GET", "/users/@me"))
         except HTTPException as exc:
-            self.token = old_token
+            # self.token = old_token
             if exc.status == 401:
                 raise LoginFailure("Improper token has been passed.") from exc
             raise
 
         return data
 
+    async def exchange_access_code(
```

Comment:
This new method and all the other OAuth2 related changes should be dropped if this PR just focuses on rate limiting.